src/config/backend/EdFi.DmsConfigurationService.Backend.Postgresql/OpenIddict/PostgresTokenManager.cs

```diff
@@ -0,0 +1,691 @@
+            {
+                return await LoadActiveSigningKeyFromCertificatesAsync();
+            }
+            else
+            {
+                return await LoadActiveSigningKeyFromDatabaseAsync();
+            }
+        }
+
+        /// <summary>
+        /// Loads signing key from X.509 certificates (existing implementation)
+        /// </summary>
+        private async Task<(SecurityKey, string)> LoadActiveSigningKeyFromCertificatesAsync()
+        {
+            if (_identityOptions.Value.UseDevelopmentCertificates)
+            {
+                var certPath = _identityOptions.Value.DevCertificatePath;
+                var certPassword = _identityOptions.Value.DevCertificatePassword;
+                X509Certificate2 cert;
+                if (!System.IO.File.Exists(certPath))
+                {
+                    using var rsa = RSA.Create(2048);
+                    var certRequest = new CertificateRequest(
+                        "CN=DevCert",
+                        rsa,
+                        System.Security.Cryptography.HashAlgorithmName.SHA256,
+                        System.Security.Cryptography.RSASignaturePadding.Pkcs1
+                    );
+                    cert = certRequest.CreateSelfSigned(
+                        DateTimeOffset.UtcNow.AddDays(-1),
+                        DateTimeOffset.UtcNow.AddYears(1)
+                    );
+                    var bytes = cert.Export(X509ContentType.Pfx, certPassword);
+                    await System.IO.File.WriteAllBytesAsync(certPath, bytes);
+                }
+                else
+                {
+                    cert = new X509Certificate2(certPath, certPassword);
+                }
+                var signingKey = new X509SecurityKey(cert);
+                return await Task.FromResult((signingKey, cert.Thumbprint));
+            }
+            else
+            {
+                // Load certificate from configured path
+                var certPath = _identityOptions.Value.CertificatePath;
+                var certPassword = _identityOptions.Value.CertificatePassword;
+                if (string.IsNullOrEmpty(certPath))
+                {
+                    throw new InvalidOperationException("CertificatePath must be set when not using development certificates.");
+                }
+                var cert = string.IsNullOrEmpty(certPassword)
+                    ? new System.Security.Cryptography.X509Certificates.X509Certificate2(certPath)
+                    : new System.Security.Cryptography.X509Certificates.X509Certificate2(certPath, certPassword);
+                var signingKey = new X509SecurityKey(cert);
+                return await Task.FromResult((signingKey, cert.Thumbprint));
+            }
+        }
+
+        /// <summary>
+        /// Loads signing key from database (OpenIddictKey table)
+        /// </summary>
+        private async Task<(SecurityKey, string)> LoadActiveSigningKeyFromDatabaseAsync()
```

Comment:
Take a look at how we "brand" strings in DMS. We'll want to do that in CMS too. Passing plain string types is like passing "object". We know semantically what our strings are. This one is a KeyId. See: src/dms/core/EdFi.DataManagementService.Core.External/Model/ReferentialId.cs and src/dms/core/EdFi.DataManagementService.Core.External/Model/TraceId.cs and src/dms/core/EdFi.DataManagementService.Core.External/Model/ContactUniqueId.cs as examples.

Reply:
I updated the method but to return an object. I'll check your example for future reference.  Is that similar to the types we used in meadowlark for strings?